examples/cat_facts/shared/src/app.rs

```diff
@@ -126,89 +127,97 @@ impl App for CatFacts {
-                    .send(Event::SetImage);
-
-                return render::render();
+                Command::all([
+                    Http::get(FACT_API_URL)
+                        .expect_json()
+                        .build()
+                        .then_send(Event::SetFact),
+                    Http::get(IMAGE_API_URL)
+                        .expect_json()
+                        .build()
+                        .then_send(Event::SetImage),
+                    render::render(),
+                ])
             }
             Event::SetFact(Ok(mut response)) => {
                 model.cat_fact = Some(response.take_body().unwrap());
 
                 caps.time.now(Event::CurrentTime);
+                Command::done()
             }
             Event::SetImage(Ok(mut response)) => {
                 model.cat_image = Some(response.take_body().unwrap());
 
                 let bytes = serde_json::to_vec(&model).unwrap();
 
-                return Command::all([
+                Command::all([
                     key_value::set(KEY, bytes).then_send(|_| Event::None),
                     render::render(),
-                ]);
+                ])
             }
             Event::SetFact(Err(_)) | Event::SetImage(Err(_)) => {
-                // TODO: Display an error
+                // handle error
+                Command::done()
             }
             Event::CurrentTime(TimeResponse::Now { instant }) => {
                 let time: DateTime<Utc> = instant.try_into().unwrap();
                 model.time = Some(time.to_rfc3339_opts(chrono::SecondsFormat::Secs, true));
 
                 let bytes = serde_json::to_vec(&model).unwrap();
 
-                return Command::all([
+                Command::all([
                     key_value::set(KEY, bytes).then_send(|_| Event::None),
                     render::render(),
-                ]);
+                ])
             }
             Event::CurrentTime(_) => panic!("Unexpected time response"),
-            Event::Restore => return key_value::get(KEY).then_send(Event::SetState),
-            Event::SetState(Ok(Some(value))) => {
-                if let Ok(m) = serde_json::from_slice::<Model>(&value) {
+            Event::Restore => key_value::get(KEY).then_send(Event::SetState),
+            Event::SetState(Ok(Some(value))) => match serde_json::from_slice::<Model>(&value) {
+                Ok(m) => {
                     *model = m;
-                    return render::render();
-                };
-            }
+                    render::render()
+                }
+                Err(_) => {
+                    // handle error
+                    Command::done()
+                }
+            },
             Event::SetState(Ok(None)) => {
                 // no state to restore
+                Command::done()
             }
             Event::SetState(Err(_)) => {
                 // handle error
+                Command::done()
```

Comment:
Nit: lots of these branches return `Command::done()` might be easier to do early returns, but I don't mind either way.

Reply:
I kinda wanted them to be markers for "something still to do here"